packages/fuels-programs/src/call_utils.rs

```diff
@@ -178,13 +192,36 @@ pub(crate) fn get_transaction_inputs_outputs(
     let outputs = chain!(
         generate_contract_outputs(num_of_contracts),
         generate_asset_change_outputs(wallet_address, asset_ids),
+        generate_custom_outputs(calls),
         extract_variable_outputs(calls),
         extract_message_outputs(calls)
     )
     .collect();
     (inputs, outputs)
 }
 
+fn generate_custom_outputs(calls: &[ContractCall]) -> Vec<Output> {
+    calls
+        .iter()
+        .flat_map(|call| call.custom_assets.iter().collect::<Vec<_>>())
```

Comment:
```suggestion
        .flat_map(|call| &call.custom_assets)
```
You can sum references to amounts later on, no need to copy them all into another collection.